File: Games/RockPaperScissors/src/main/kotlin/jokatu/game/games/rockpaperscissors/game/RockPaperScissorsGameFactory.java
package jokatu.game.games.rockpaperscissors.game;

import jokatu.game.AbstractGameFactory;
import jokatu.game.GameID;
import org.jetbrains.annotations.NotNull;
import org.springframework.stereotype.Component;

@Component
public class RockPaperScissorsGameFactory extends AbstractGameFactory<RockPaperScissorsGame> {

	@NotNull
	@Override
	protected RockPaperScissorsGame produce(@NotNull GameID gameID, @NotNull String creatorName) {
		return new RockPaperScissorsGame(gameID);
	}
}

File: Games/RockPaperScissors/src/main/kotlin/jokatu/game/games/rockpaperscissors/game/RockPaperScissorsGameFactory.kt
package jokatu.game.games.rockpaperscissors.game

import jokatu.game.AbstractGameFactory
import jokatu.game.GameID
import org.springframework.stereotype.Component

@Component
class RockPaperScissorsGameFactory : AbstractGameFactory<RockPaperScissorsGame>() {

	override fun produce(gameID: GameID, creatorName: String): RockPaperScissorsGame {
		return RockPaperScissorsGame(gameID)
	}
}
